bundles/CoreBundle/src/EventListener/Frontend/ContentTemplateListener.php

```diff
@@ -91,14 +90,18 @@ public function onKernelView(ViewEvent $event): void
         );
     }
 
-    private function resolveParameters(ControllerArgumentsEvent $event, ?array $vars): array
+    private function resolveParameters(ViewEvent $event, ?array $vars): array
     {
-        $parameters = $event->getNamedArguments();
+        $controllerArguments = $event->controllerArgumentsEvent?->getNamedArguments() ?? [];
+        $controllerResults = is_array($event->getControllerResult()) ? $event->getControllerResult() : [];
 
-        if (null !== $vars) {
-            $parameters = array_intersect_key($parameters, array_flip($vars));
-        }
+        $mergedArray = array_merge(array_keys($controllerArguments), array_keys($controllerResults), array_keys($vars));
+        $duplicateKeys = array_unique(array_diff_assoc($mergedArray, array_unique($mergedArray)));
 
-        return $parameters;
+        if (empty($duplicateKeys)) {
+            return array_merge($controllerArguments, $controllerResults, $vars);
+        } else {
+            throw new \Exception('Duplicate keys found: '.implode(', ', array_values($duplicateKeys)).'. Please use unique names for your controller arguments, controller results and template variables.');
+        }
```

Comment:
I think this would be more readable
```suggestion
        if ($duplicateKeys) {
            throw new \Exception('Duplicate keys found: '.implode(', ', array_values($duplicateKeys)).'. Please use unique names for your controller arguments, controller results and template variables.');
        }

        return array_merge($controllerArguments, $controllerResults, $vars);
```